.github/workflows/graphics_tests.yml

```diff
@@ -47,9 +47,9 @@ jobs:
         uses: actions/upload-artifact@v3
         if: always()
         with:
-          name: test_results
+          name: test_results_${{ matrix.device }}
```

Comment:
It seems we need to avoid a conflict regarding the artifact's name
https://github.com/actions/upload-artifact?tab=readme-ov-file#not-uploading-to-the-same-artifact

Reply:
👀

Reply:
@utzcoz 🙆 
I think it's okay for other parts of this repository.
<img width="585" alt="image" src="https://github.com/robolectric/robolectric/assets/1386930/618f5e20-6c4f-4ef3-8529-3ae9f864fbf0">

* codeql.yml doesn't use upload-artifact.
* test.yml 1 seems to use the matrix name appropriately.
`name: test_results_${{ matrix.api-versions }}`
* test.yml 2 seems to use the matrix name appropriately.
`name: test-results-${{ matrix.api-level }}-${{ steps.determine-target.outputs.TARGET }}-${{ matrix.shard }}`